### PR TITLE
AGD-1300 Add mechanism for common package directory similar to common definition directory alternative

### PR DESCRIPTION
**Pull request by saintentropy**
### Purpose

The purpose of this PR is to add a mechanism to allow a common package installation directory similar to the common custom node directly already supported in the PathManager.  The common package directly will allow for the installation of packages that are available to all users on a specific machine. This common directory will supplement the current default user package and definitions directories located in user/appdata/...  

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner @QilongTang 

### FYIs

@sm6srw @Dewb

### CODE REVIEW

File: src/DynamoCore/Configuration/PathManager.cs [hunk 1]
```diff
@@ -95,6 +96,11 @@ private IEnumerable<string> RootDirectories
             get { return Preferences != null ? Preferences.CustomPackageFolders : rootDirectories; }
         }
 
+        internal string CommonPackageDirectory
```
Comment (mjkkirschner) [inline src/DynamoCore/Configuration/PathManager.cs]: please add a summary even though internal.

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 2]
```diff
@@ -124,5 +132,46 @@ private void Log(string msg)
         /// An event that is raised when an extension starts loading.
         /// </summary>
         public event Action<IExtension> ExtensionLoading;
+
+        internal List<string> directoriesToVerifyCertificates = new List<string>();
```
Comment (QilongTang) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: `DirectoriesToVerifyCertificates` thx!

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 3]
```diff
@@ -124,5 +132,46 @@ private void Log(string msg)
...
+
+        private static bool CheckExtensionCertificates(ExtensionDefinition viewExtension)
```
Comment (QilongTang) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: Would you add comments for this function for future reference? e.g. why we need this, what case we prevent.

File: src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs [hunk 4]
```diff
@@ -97,5 +106,46 @@ private void Log(string msg)
         /// An event that is raised when an extension starts loading.
         /// </summary>
         public event Action<IViewExtension> ExtensionLoading;
+
+        internal  List<string> directoriesToVerifyCertificates = new List<string>();
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs]: `DirectoriesToVerifyCertificates` thx!

File: src/DynamoCore/Configuration/PathManager.cs [hunk 5]
```diff
@@ -72,6 +72,7 @@ class PathManager : IPathManager
         private readonly string commonDataDir;
 
         private readonly string commonDefinitions;
+        private readonly string commonPackages;
```
Comment (mjkkirschner) [inline src/DynamoCore/Configuration/PathManager.cs]: @saintentropy are you planning to add a TODO item to PathManager interface? or Add a new interface containing this item that we can merge back together later? This improves our chances of actually folding this back into any new apis in new releases.

Comment (saintentropy) [reply src/DynamoCore/Configuration/PathManager.cs]: Will do

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 6]
```diff
@@ -124,5 +132,46 @@ private void Log(string msg)
...
+                asm = Assembly.ReflectionOnlyLoadFrom(viewExtension.AssemblyPath);
+            }
+            catch
+            {
+                throw new Exception(String.Format(
+                    "An extension called {0} found at {1} has a dll which could not be loaded.  Ignoring it.",
+                    viewExtension.TypeName, viewExtension.AssemblyPath));
+            }
+
+            //Verify teh node libarary has a verified signed certificate
```
Comment (mjkkirschner) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: spelling (the)

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 7]
```diff
@@ -124,5 +132,46 @@ private void Log(string msg)
...
+
+            //Verify that you can load the node library assembly into a Reflection only context
+            Assembly asm;
+            try
+            {
+                asm = Assembly.ReflectionOnlyLoadFrom(viewExtension.AssemblyPath);
+            }
+            catch
+            {
+                throw new Exception(String.Format(
```
Comment (mjkkirschner) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: do these exceptions end up in the console only or are they more visible to the user?

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 8]
```diff
@@ -124,5 +132,46 @@ private void Log(string msg)
...
+                    "An extension called {0} found at {1} has a dll which could not be loaded.  Ignoring it.",
+                    viewExtension.TypeName, viewExtension.AssemblyPath));
+            }
+
+            //Verify teh node libarary has a verified signed certificate
+            var cert = asm.Modules.FirstOrDefault()?.GetSignerCertificate();
+            if (cert != null)
+            {
+                var cert2 = new System.Security.Cryptography.X509Certificates.X509Certificate2(cert);
```
Comment (mjkkirschner) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: have you verified this works on linux/mac under mono? .net Core? - if not I wonder if this flow should be disabled if running on non windows platform?

File: src/DynamoCore/Models/DynamoModel.cs [hunk 9]
```diff
@@ -690,7 +694,7 @@ protected DynamoModel(IStartConfiguration config)
             NodeFactory = new NodeFactory();
             NodeFactory.MessageLogged += LogMessage;
 
-            extensionManager = new ExtensionManager();
+            extensionManager = new ExtensionManager(new[] { PathManager.CommonDataDirectory });
```
Comment (mjkkirschner) [inline src/DynamoCore/Models/DynamoModel.cs]: maybe a comment here...

File: src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs [hunk 10]
```diff
@@ -97,5 +106,46 @@ private void Log(string msg)
...
+                throw new Exception(String.Format(
+                    "A view extension called {0} found at {1} has a dll which could not be loaded.  Ignoring it.",
+                    viewExtension.TypeName, viewExtension.AssemblyPath));
+            }
+
+            //Verify teh node libarary has a verified signed certificate
+            var cert = asm.Modules.FirstOrDefault()?.GetSignerCertificate();
+            if (cert != null)
+            {
+                var cert2 = new System.Security.Cryptography.X509Certificates.X509Certificate2(cert);
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs]: this code seems exactly the same - what am I missing? can't we use the version in dynamo core?

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 11]
```diff
@@ -98,6 +98,8 @@ public DynamoView(DynamoViewModel dynamoViewModel)
 
             tabSlidingWindowStart = tabSlidingWindowEnd = 0;
 
+            viewExtensionManager = new ViewExtensionManager(new[] {dynamoViewModel.Model.PathManager.CommonDataDirectory });
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs]: comment

File: src/DynamoPackages/PackageLoader.cs [hunk 12]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
+                }
+                catch
+                {
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(
+                            "A package called {0} found at {1} has dlls defined in the package manifest which could not be loaded.  Ignoring it.",
+                            discoveredPkg.Name, discoveredPkg.RootDirectory));
+                }
+
+                //Verify teh node libarary has a verified signed certificate
```
Comment (mjkkirschner) [inline src/DynamoPackages/PackageLoader.cs]: spelling the

File: src/DynamoPackages/PackageLoader.cs [hunk 13]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
+                        continue;
+                    }
+                }
+
+                throw new LibraryLoadFailedException(directory,
+                    String.Format("A package called {0} found at {1} did not have signed dll files.  Ignoring it.",
+                        discoveredPkg.Name, discoveredPkg.RootDirectory));
+            }
+
+            discoveredPkg.HasSignedEntryPoints = true;
```
Comment (mjkkirschner) [inline src/DynamoPackages/PackageLoader.cs]: what conditions will make this true?

File: src/DynamoPackages/PackageLoader.cs [hunk 14]
```diff
@@ -75,12 +75,29 @@ public string DefaultPackagesDirectory
...
             : this(new[] { overridePackageDirectory })
         {
         }
 
         public PackageLoader(IEnumerable<string> packagesDirectories)
+        {
+            InitializePackageDirectories(packagesDirectories);
+        }
+
+        public PackageLoader(IEnumerable<string> packagesDirectories, IEnumerable<string> packageDirectoriesToVerify)
```
Comment (mjkkirschner) [inline src/DynamoPackages/PackageLoader.cs]: please add a summary for this constructor

File: src/DynamoPackages/PackageLoader.cs [hunk 15]
```diff
@@ -75,12 +75,29 @@ public string DefaultPackagesDirectory
...
+        {
+            InitializePackageDirectories(packagesDirectories);
+
+            if (packageDirectoriesToVerify == null)
+                throw new ArgumentNullException("packageDirectoriesToVerify");
```
Comment (mjkkirschner) [inline src/DynamoPackages/PackageLoader.cs]: nameof?

File: src/DynamoCore/Extensions/ExtensionLoader.cs (hunk 7 above)
Comment (saintentropy) [reply src/DynamoCore/Extensions/ExtensionLoader.cs]: I don't know yet.  I will test and verify

File: src/DynamoPackages/PackageLoader.cs [hunk 16]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
             return null;
         }
 
+        private static void CheckPackageNodeLibraryCertificates(string directory, Package discoveredPkg)
```
Comment (mjkkirschner) [inline src/DynamoPackages/PackageLoader.cs]: this is a giant method - please add a summary of what it does.

File: src/DynamoPackages/PackageLoader.cs (hunk 13 above)
Comment (saintentropy) [reply src/DynamoPackages/PackageLoader.cs]: You get here when you have looped through each file in the node_libraries and verified it has a cert that is verified.  Everywhere else you hit a exception

File: src/DynamoPackages/Package.cs [hunk 17]
```diff
@@ -294,6 +306,27 @@ internal IEnumerable<PackageAssembly> EnumerateAssembliesInBinDirectory()
             return assemblies;
         }
 
+        private static bool IsFileInManifestNodeLibraries(IEnumerable<string> nodeLibraries, string filename, string path)
```
Comment (mjkkirschner) [inline src/DynamoPackages/Package.cs]: I don't understand the name of this function.

File: src/DynamoPackages/Package.cs [hunk 18]
```diff
@@ -294,6 +306,27 @@ internal IEnumerable<PackageAssembly> EnumerateAssembliesInBinDirectory()
...
+                try
+                {
+                    var name = new AssemblyName(manifestFile).Name + ".dll";
+                    if (name == filename)
+                    {
+                        return true;
+                    }
+                }
+                catch
+                {;
```
Comment (mjkkirschner) [inline src/DynamoPackages/Package.cs]: dangling semicolon?

File: src/DynamoPackages/Package.cs [hunk 19]
```diff
@@ -294,6 +306,27 @@ internal IEnumerable<PackageAssembly> EnumerateAssembliesInBinDirectory()
             return assemblies;
         }
 
+        private static bool IsFileInManifestNodeLibraries(IEnumerable<string> nodeLibraries, string filename, string path)
+        {
+            foreach (var manifestFile in nodeLibraries)
```
Comment (mjkkirschner) [inline src/DynamoPackages/Package.cs]: this name is confusing to me - what about nodeLibraryAssemblyName?

File: src/DynamoPackages/PackageLoader.cs [hunk 20]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
+            var dllfiles = new System.IO.DirectoryInfo(discoveredPkg.BinaryDirectory).EnumerateFiles("*.dll");
+            if (discoveredPkg.Header.node_libraries.Count() == 0 && dllfiles.Count() != 0)
+            {
+                throw new LibraryLoadFailedException(directory,
+                    String.Format(
+                        "A package called {0} found at {1} includes dll files but none are defined in node libraries in the package manifest.  Ignoring it.",
+                        discoveredPkg.Name, discoveredPkg.RootDirectory));
+            }
+
+            foreach (var manifestFile in discoveredPkg.Header.node_libraries)
```
Comment (mjkkirschner) [inline src/DynamoPackages/PackageLoader.cs]: same with the name here - unless I am misunderstanding - personally I would get this confused with the manifest file itself - like the header file or the extensionManfiest

File: src/DynamoPackages/PackageLoader.cs [hunk 21]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
+            {
+
+                //Try to get the assembly name from the manifest file
+                string filename;
+                try
+                {
+                    filename = new AssemblyName(manifestFile).Name + ".dll";
+                }
+                catch
```
Comment (mjkkirschner) [inline src/DynamoPackages/PackageLoader.cs]: usually it's good to catch the exception and log it as well.

File: src/DynamoPackages/PackageLoader.cs [hunk 22]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
+                        String.Format(
+                            "A package called {0} found at {1} is missing dlls which are defined in the package manifest.  Ignoring it.",
+                            discoveredPkg.Name, discoveredPkg.RootDirectory));
+                }
+
+                //Verify that you can load the node library assembly into a Reflection only context
+                Assembly asm;
+                try
+                {
+                    asm = Assembly.ReflectionOnlyLoadFrom(filename);
```
Comment (mjkkirschner) [inline src/DynamoPackages/PackageLoader.cs]: I really hope this doesn't cause issues with loading the assembly later if it's already loaded here?

Comment (mjkkirschner) [reply src/DynamoPackages/PackageLoader.cs]: can we verify this somehow?

**Comment by mjkkirschner** [comment]
@saintentropy it looks good - a few comments...
also -

it needs tests, here are some I think would be good:
* try it on a package with cert - what happens?
* try it on a package with a cert which has been modified - what happens?
* load a dll into reflection only context, then load a package using it - any problems ? This could also be solved with some research I guess? - am I interpreting this correctly that it will be loaded into both contexts after your PR if it requires a cert?
* x502 methods on linux mac? Are they supported? seems on .net core at least: https://github.com/dotnet/announcements/issues/21
* Have you considered the case that someone installs refinery somewhere else and leaves it this location? Or has Refinery already installed? Do you want to avoid loading the second package? Perhaps test this if a test does not already exist?

File: src/DynamoPackages/PackageLoader.cs [hunk 23]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
+                //Try to get the assembly name from the manifest file
+                string filename;
+                try
+                {
+                    filename = new AssemblyName(manifestFile).Name + ".dll";
+                }
+                catch
+                {
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(
```
Comment (aparajit-pratap) [inline src/DynamoPackages/PackageLoader.cs]: This string should be added to resource files.

File: src/DynamoPackages/PackageLoader.cs [hunk 24]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
+
+
+                //Verify the node library exists in the package bin directory
+                var filepath = Path.Combine(discoveredPkg.BinaryDirectory, filename + ".dll");
+                if (!File.Exists(filepath))
+                {
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(
+                            "A package called {0} found at {1} is missing dlls which are defined in the package manifest.  Ignoring it.",
+                            discoveredPkg.Name, discoveredPkg.RootDirectory));
```
Comment (aparajit-pratap) [inline src/DynamoPackages/PackageLoader.cs]: Add to string resources.

File: src/DynamoPackages/PackageLoader.cs [hunk 25]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
+                try
+                {
+                    asm = Assembly.ReflectionOnlyLoadFrom(filename);
+                }
+                catch
+                {
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(
+                            "A package called {0} found at {1} has dlls defined in the package manifest which could not be loaded.  Ignoring it.",
+                            discoveredPkg.Name, discoveredPkg.RootDirectory));
```
Comment (aparajit-pratap) [inline src/DynamoPackages/PackageLoader.cs]: Same here.

File: src/DynamoPackages/PackageLoader.cs [hunk 26]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
+                    var cert2 = new System.Security.Cryptography.X509Certificates.X509Certificate2(cert);
+                    if (cert2.Verify())
+                    {
+                        continue;
+                    }
+                }
+
+                throw new LibraryLoadFailedException(directory,
+                    String.Format("A package called {0} found at {1} did not have signed dll files.  Ignoring it.",
+                        discoveredPkg.Name, discoveredPkg.RootDirectory));
```
Comment (aparajit-pratap) [inline src/DynamoPackages/PackageLoader.cs]: And here.

File: src/DynamoPackages/PackageLoader.cs [hunk 27]
```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
...
 
+        private static void CheckPackageNodeLibraryCertificates(string directory, Package discoveredPkg)
+        {
+            var dllfiles = new System.IO.DirectoryInfo(discoveredPkg.BinaryDirectory).EnumerateFiles("*.dll");
+            if (discoveredPkg.Header.node_libraries.Count() == 0 && dllfiles.Count() != 0)
+            {
+                throw new LibraryLoadFailedException(directory,
+                    String.Format(
+                        "A package called {0} found at {1} includes dll files but none are defined in node libraries in the package manifest.  Ignoring it.",
+                        discoveredPkg.Name, discoveredPkg.RootDirectory));
```
Comment (aparajit-pratap) [inline src/DynamoPackages/PackageLoader.cs]: Add to string resource.

File: src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs [hunk 28]
```diff
@@ -97,5 +106,46 @@ private void Log(string msg)
...
+                if (cert2.Verify())
+                {
+                    return true;
+                }
+            }
+
+            throw new Exception(String.Format(
+                "A view extension called {0} found at {1} did not have a signed certificate.  Ignoring it.",
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs]: Add to string resources.

File: src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs [hunk 29]
```diff
@@ -97,5 +106,46 @@ private void Log(string msg)
...
+            Assembly asm;
+            try
+            {
+                asm = Assembly.ReflectionOnlyLoadFrom(viewExtension.AssemblyPath);
+            }
+            catch
+            {
+                throw new Exception(String.Format(
+                    "A view extension called {0} found at {1} has a dll which could not be loaded.  Ignoring it.",
+                    viewExtension.TypeName, viewExtension.AssemblyPath));
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs]: Add to resx files.

File: src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs [hunk 30]
```diff
@@ -97,5 +106,46 @@ private void Log(string msg)
...
+        internal  List<string> directoriesToVerifyCertificates = new List<string>();
+
+        private static bool CheckExtensionCertificates(ViewExtensionDefinition viewExtension)
+        {
+            //Verify the node library exists in the package bin directory
+            if (!File.Exists(viewExtension.AssemblyPath))
+            {
+                throw new Exception(String.Format(
+                        "A view extension called {0} found at {1} is missing dlls which are defined in the view extension definition.  Ignoring it.",
+                        viewExtension.TypeName, viewExtension.AssemblyPath));
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs]: Must be string resource.

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 31]
```diff
@@ -124,5 +132,46 @@ private void Log(string msg)
...
+                if (cert2.Verify())
+                {
+                    return true;
+                }
+            }
+
+            throw new Exception(String.Format(
+                "An extension called {0} found at {1} did not have a signed certificate.  Ignoring it.",
+                viewExtension.TypeName, viewExtension.AssemblyPath));
```
Comment (aparajit-pratap) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: string resource

File: src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs (hunk 10 above)
Comment (saintentropy) [reply src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs]: Not sure what you mean?  That there is similar code here and in the package loading path?

File: src/DynamoPackages/PackageLoader.cs (hunk 15 above)
Comment (saintentropy) [reply src/DynamoPackages/PackageLoader.cs]: Not sure what you mean

Comment (mjkkirschner) [reply src/DynamoPackages/PackageLoader.cs]: use `nameof(packagesDirectories)`

File: src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs (hunk 10 above)
Comment (mjkkirschner) [reply src/DynamoCoreWpf/Extensions/ViewExtensionLoader.cs]: yes, there's a bunch of duplicate code here that can get out of sync.

File: src/DynamoCore/Extensions/ExtensionDefinition.cs [hunk 32]
```diff
@@ -10,5 +10,7 @@ public class ExtensionDefinition
 
         [DataMember]
         public string TypeName { get; set; }
+
+        public bool RequiresSignedEntryPoint { get; internal set; } = false;
```
Comment (mjkkirschner) [inline src/DynamoCore/Extensions/ExtensionDefinition.cs]: needs a summary

Comment (mjkkirschner) [reply src/DynamoCore/Extensions/ExtensionDefinition.cs]: hmmmmm - does this effect existing extensions and their automatic serialization and deserialization?
Can you show a sample?

Comment (mjkkirschner) [reply src/DynamoCore/Extensions/ExtensionDefinition.cs]: doesn't this seem kind of fragile?...

File: src/DynamoCoreWpf/Extensions/ViewExtensionDefinition.cs [hunk 33]
```diff
@@ -10,5 +10,7 @@ internal class ViewExtensionDefinition
 
         [DataMember]
         public string TypeName { get; set; }
+
+        public bool RequiresSignedEntryPoint { get; internal set; } = false;
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Extensions/ViewExtensionDefinition.cs]: comments etc

File: src/DynamoPackages/Package.cs [hunk 34]
```diff
@@ -139,6 +140,8 @@ internal IEnumerable<Assembly> NodeLibraries
         /// </summary>
         public PackageUploadRequestBody Header { get; internal set; }
 
+        public bool RequiresSignedEntryPoints { get; internal set; }
```
Comment (mjkkirschner) [inline src/DynamoPackages/Package.cs]: does this need to be public? Does this effect the package header? Summary if it needs to be public.

Comment (mjkkirschner) [reply src/DynamoPackages/Package.cs]: If it effects header then we need to consider if this effects package upload/download?

File: src/DynamoCore/Extensions/ExtensionDefinition.cs (hunk 32 above)
Comment (saintentropy) [reply src/DynamoCore/Extensions/ExtensionDefinition.cs]: hmmm... Didn't realize we automatically deserialize.  But yes that would be not good

File: src/DynamoCore/Configuration/PathManager.cs (hunk 1 above)
Comment (saintentropy) [reply src/DynamoCore/Configuration/PathManager.cs]: Done and TODO

File: src/DynamoCore/Extensions/ExtensionDefinition.cs (hunk 32 above)
Comment (saintentropy) [reply src/DynamoCore/Extensions/ExtensionDefinition.cs]: We don't ever serialize or deserialize to this class automatically.  We had build this class object always.  But regardless, this does not have to be a public property.  I will move it and the ViewExtension version to internal

File: src/DynamoPackages/Package.cs (hunk 34 above)
Comment (saintentropy) [reply src/DynamoPackages/Package.cs]: Will move this to internal as well

File: src/DynamoCore/Extensions/ExtensionDefinition.cs (hunk 32 above)
Comment (mjkkirschner) [reply src/DynamoCore/Extensions/ExtensionDefinition.cs]: hmm - I see - Wonder why the dataMember attributes were used.

So you don't intend for this property to be deserialized?

**Comment by mjkkirschner** [comment]
@saintentropy the tests look good - but it would be useful to have the source for the dlls in the repo I guess... though I *guess* we can always decompile them...

**Comment by saintentropy** [comment]
I guess I thought the sample packages were in the repo.  I just signed the existing package sample

**Pull request merged**